### ISSUE TITLE
WSL Hyper-V-VmSwitch restarts network every few minutes causing sound stuttering on host machine

### ISSUE THREAD

**Issue by omgmajk** [microsoft/WSL#7178]
### Windows Build Number

Version 10.0.19043.1110

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.4.72-microsoft-standard-WSL2

### Distro Version

Kali

### Repro Steps

Boot machine, sit and wait. No other steps are necessary for this to happen on my machine. Not sure what I can do to mitigare this. 

### Expected Behavior

WSL2 should not restart network interface Hyper-V-VmSwitch every 2-10 minutes.

### Actual Behavior

Every 2-10 minutes (sometimes a break of up to half an hour) Hyper-V-VmSwitch restarts it's network interface and logs it to windows logs, which causes sound stuttering on my usb headphones.

`Networking driver in Virtual Machine is loaded and the protocol version is negotiated to the most recent version (Virtual machine ID 4E28C58C-BE6A-451D-A5E8-E944A5756E4D).

NIC 4E28C58C-BE6A-451D-A5E8-E944A5756E4D--FC785225-9131-5661-AC0C-3A157C61AE15 (Friendly Name: ) successfully connected to port 32097E04-6456-4FBB-AA70-A91218419EFB (Friendly Name: 5E03A28A-6836-45A1-AE15-0C24DBFDE003) on switch E4E1D1EB-6009-48AD-8CC8-E195490F0524(Friendly Name: WSL).

The operation 'Create' succeeded on nic 4E28C58C-BE6A-451D-A5E8-E944A5756E4D--FC785225-9131-5661-AC0C-3A157C61AE15 (Friendly Name: ). Port 32097E04-6456-4FBB-AA70-A91218419EFB (Friendly Name: 5E03A28A-6836-45A1-AE15-0C24DBFDE003) successfully created on switch E4E1D1EB-6009-48AD-8CC8-E195490F0524 (Friendly Name: WSL).

The operation 'Delete' succeeded on nic 0E162F26-4FC8-452F-972A-43E573F7DEFA--FC785225-9131-5661-AC0C-3A157C61AE15 (Friendly Name: ).`

![hyperv](https://user-images.githubusercontent.com/10681392/125698696-4fea107d-94db-4cbc-a86f-607296a91298.png)

### Diagnostic Logs

[diagnostics.zip](https://github.com/microsoft/WSL/files/6819180/diagnostics.zip)

**Comment by craigloewen-msft** [comment]
Could you please file a feedback hub item for this with the profile as "Audio" related instead of WSL? Starting this VM switch shouldn't make your audio stutter and I would want to triage that to the audio team!

You can find instructions of how to use Feedback Hub logs at the comment below ( /logs ) but instead of parenting it to WSL could you parent it to the audio team?

To help get you unblocked in the meantime, could you try and [clear recent files from File Explorer](https://www.onmsft.com/how-to/how-to-clear-recent-files-and-folders-from-windows-10-file-explorer#:~:text=1%20Launch%20File%20Explorer.%202%20Click%20File%20%3E,%E2%80%9CClear%E2%80%9D%20button%20under%20the%20%E2%80%9CPrivacy%E2%80%9D%20section.%20See%20More). I'm wondering if you have some WSL files there which are getting queried and starting up the WSL 2 VM which is why this might be happening so frequently.

**Comment by omgmajk** [comment]
> Could you please file a feedback hub item for this with the profile as "Audio" related instead of WSL? Starting this VM switch shouldn't make your audio stutter and I would want to triage that to the audio team!
> 
> You can find instructions of how to use Feedback Hub logs at the comment below ( /logs ) but instead of parenting it to WSL could you parent it to the audio team?
> 
> To help get you unblocked in the meantime, could you try and [clear recent files from File Explorer](https://www.onmsft.com/how-to/how-to-clear-recent-files-and-folders-from-windows-10-file-explorer#:~:text=1%20Launch%20File%20Explorer.%202%20Click%20File%20%3E,%E2%80%9CClear%E2%80%9D%20button%20under%20the%20%E2%80%9CPrivacy%E2%80%9D%20section.%20See%20More). I'm wondering if you have some WSL files there which are getting queried and starting up the WSL 2 VM which is why this might be happening so frequently.

Hi, I already tested this method of clearing out my recent files as I saw that in another issue before I opened this one. I will quote myself from SuperUser:

> I try to run a wsl --shutdown or wsl -t kali-linux but it keeps spinning up again, and even if it's running it's still doing these things and causing sound issues. I added info to a github issue on this matter and they responded to another person stating that if you had files relating to wsl in your recent files folder it would spin up automatically, so I cleared out my recent files and it spins up anyways.
>
>Technically, I just need help getting rid of my sound problems. I use kali in windows so often it would be a real shame to get rid of it but I can't live with sound stuttering every two minutes either.
>
>LatencyMon complained about various things but the images in the microsoft thread above is with a bunch of processes, including spotify or youtube running while it was doing it's thing so they are not too accurate and I have already re-installed and upgraded/downgraded all drivers relating to sound, network and graphics about four times now.
>
>I also read somewhere here that this issue can be related to windows Dns Service not running but it's already running.

[SuperUser thread](https://superuser.com/questions/1662608/wsl-hyper-v-vmswitch-connects-reconnects-does-something-with-my-networking-eve)

[Microsoft Q&A Forum with more images](https://docs.microsoft.com/en-us/answers/questions/466983/sound-stutterslagspops-on-host-machine-whenever-hy.html)

I will have to re-install WSL2 and the virtualization environment and do more testing but I finally caved in an went with VMWare, which is a shame as I really liked using Kali more native in windows. I will try to file a feedback hub item with this as soon as I can.

**Comment by craigloewen-msft** [comment]
Ok great! Please feel free to file something as soon as you're able.

**Comment by sasjafor** [comment]
I'm also experiencing this issue. EventViewer shows the same events from HyperV about restarting the networking driver. And running LatencyMon shows that tcpip.sys had a highest reported DPC routine execution time of 23957 us (microseconds).

I'm running Windows 10 21H2 with build version 19044.1526 and the Hyper-V feature itself is not activated. Only Virtual Machine Platform, the Windows Hypervisor Platform & the WSL feature are enabled.

Here is a link to a feedback hub issue I created https://aka.ms/AAfyv34 (without recording because it's hard to record something that happens randomly a few times per hour of course).

**Comment by ikifar2012** [comment]
I am having the same issue every time I type `wsl --shutdown` my audio stutters massively

**Comment by sasjafor** [comment]
@ikifar2012 Are you perhaps using a VPN with split tunneling or something similar? That was the solution for me, as soon as I turned that feature off the problem was gone.

**Comment by ikifar2012** [comment]
> @ikifar2012 Are you perhaps using a VPN with split tunneling or something similar? That was the solution for me, as soon as I turned that feature off the problem was gone.

Nope, but I did have a ton of VMWare network adapters, I have now disabled them but I doubt that is gonna help, but I will report back if it does

**Comment by ticinum-aerospace** [comment]
Hi, I've the same issue as well, it occours with realtek wms and asio drivers as well. Very very annoying.

**Comment by Feirell** [comment]
@Omny that might have been it! I had a link in my windows user directory, which was a test to have a direct link to the wsl directory.

I removed it and have not experienced it since. The only thing which makes me wonder, is that I had this link for some time now but did not had the issue since. Any ways it seems to be gone!

**Comment by JosefSaltz** [comment]
> I had this problem when I pinned the WSL folder in Quick acces in Explorer. Explorer constantly accessed this folder and started/shut down the WSL virtual machine, which created friezes and sound stutters every time.

Novel idea, I'm going to give this a try. I notice the latency issues occur mostly after running `WSL.exe --shutdown`

**Comment by dajhorn** [comment]
This bug persists into Windows 11 Version 21H2 Build 22000.1042.

Thousands of messages like this are spammed into the system event log each hour:
```
The operation 'Delete' succeeded on nic E1238137-8080-481C-A23C-E18812B814F6--2AAED518-1EFD-4E39-A501-66A3268C9AD6 (Friendly Name: ), Instance Id {00000000-0000-0000-0000-000000000000}.

Networking driver in Virtual Machine is loaded and the protocol version is negotiated to the most recent version (Virtual machine ID E1238137-8080-481C-A23C-E18812B814F6).

NIC E1238137-8080-481C-A23C-E18812B814F6--2AAED518-1EFD-4E39-A501-66A3268C9AD6 (Friendly Name: ) successfully connected to port 4F1EF933-D638-4CA6-B017-CEC1A72B7352 (Friendly Name: 2AAED518-1EFD-4E39-A501-66A3268C9AD6) on switch B8D4D282-A208-47BE-B0B0-65F38BC5CA85(Friendly Name: WSL).

Port 4F1EF933-D638-4CA6-B017-CEC1A72B7352 (Friendly Name: 2AAED518-1EFD-4E39-A501-66A3268C9AD6) successfully created on switch B8D4D282-A208-47BE-B0B0-65F38BC5CA85 (Friendly Name: WSL).
```

**Comment by louisthomaspro** [comment]
Same issue on my side. I got those Hyper-V-VmSwitch logs every 30 minutes. It wakes up my laptop from sleep mode.
The issue started when I installed WSL2.
Windows 11 Version 10.0.22621 Build 22621

**Comment by louisthomaspro** [comment]
I fixed the issue by going on :
Folder options > General > Privacy
I disabled **Show frequently used folder**, **Show recently used files** and clicked on the clear button (clear file explorer history)

Hope it helps !

**Comment by Varcain** [comment]
I can confirm this on my machine. At first I thought it was a problem with my USB Audio Interface (Steinberg UR22mkII). It was randomly disconnecting and then reconnecting from USB. After googling "Steinberg UR22mkII USB dropouts" found no solution. Next, I disconnected the audio i/f and switched to built-in Realtek card. It also manifested the issue by causing weird audio drops from time to time. After hours of trial and error I've found what fixes the issue: starting WSL terminal.
Terminating WSL caused the issue to reproduce again. This has lead me to this thread. My setup:
- Hyper-V VM with WLAN bridge setup
- WSL2

Root cause is *not* an audio issue. Audio issues seem to be caused by high latency spike caused by VM related process which are gone after opening WSL shell. Logs also show network interface restarting for me.

**Comment by ikifar2012** [comment]
Any updates on this issue? I really liked WSL but this issue has prevented me from using it

**Comment by ti-mo** [comment]
Damn, found this issue too late, just reinstalled because of this. Started happening a few days ago, gives me a heart attack everytime it happens. Could I provide more info to get this moving? Kind of makes WSL unusable for me.

**Comment by scristall** [comment]
Also having this issue. Every 1-2 minutes, whole machine was stuttering, and found the WSL related logs in the Event Viewer.

Keeping a WSL2 terminal open in the background has been working for me as a workaround in the time being.

**Comment by marinofranz** [comment]
Having this issue as well. There are a few audio issues like stuttering & audio cutouts. I have only installed WSL2 today and started having this issue. I really would hate seeing myself uninstalling WSL just because of this issue.

**Comment by Theliel** [comment]
I ran into the problem today by chance, although from what I can see in the event viewer it has been happening for a long time. 

The solution in my case was similar. By default, a shortcut to the linux FS its added to Explorer (at the bottom, sidebar), which causes WSL to run again in the background every few seconds/minutes, creating a loop. This not only causes flooding Event logging, in my case it also caused me to lose focus of other windows from time to time, including bringing Windows Explorer to the foreground suddenly (Still in testing, but it seems that it has also been solved)

In my case, solution was as simple as removing the shortcut. Regedit:

HKEY_CLASSES_ROOT\CLSID\{B2B4A4D1-2754-4140-A2EB-9A76D9D7CDC6}

Removing the dword System.IsPinnedToNameSpaceTree was enough. You may need to modify the permissions of the key, by default they are read-only and the owner TrustedInstaller

Close all open explorer windows and solved. Everything continues to work perfectly.

**Comment by polwel** [comment]
Having the same issue. The flood of events in the event viewer still keeps coming after I disabled the WSL service, so I am wondering if WSL is to blame at all, at least in my case.

**Comment by ikifar2012** [comment]
> Having the same issue. The flood of events in the event viewer still keeps coming after I disabled the WSL service, so I am wondering if WSL is to blame at all, at least in my case.

Its definitely a HyperV issue, I have been using HyperV now as well and every single time I start and stop a VM I get a stutter and sometimes I have to restart the Windows Audio service for it to stop stuttering

**Comment by tadghh** [comment]
This issue is consistent across pcs when using a focusrite solo 3rd gen. Not sure what everyone else is using

**Comment by overherz** [comment]
I have the same problem, I've been using wsl for 2 years and this has never happened before
I don't know what led to this
windows 11 24H2
WSL: 2.3.26.0
This manifests itself as the sound of connecting some device
Just step away from the computer and after half an hour this sound appears every few seconds

**Comment by ikifar2012** [comment]
Just filed feedback under Hyper-V on the feedback app, please upvote, comment and add diagnostics to hopefully get someone at Microsoft to acknowledge this issue https://aka.ms/AAx85xw

<img width="3170" height="1234" alt="Image" src="https://github.com/user-attachments/assets/f37d6316-0421-4605-9e31-7131fc577957" />

**Comment by ikifar2012** [comment]
Experienced the issue 3 times today, super loud, super annoying, only restarting the audio service could get it to stop

**Comment by ikifar2012** [comment]
Still happening daily
